src/manager.rs

```diff
@@ -109,15 +109,18 @@ impl<S: EventSubscriber> EventManager<S> {
     /// Wait for events for a maximum timeout of `miliseconds`. Dispatch the events to the
     /// registered signal handlers.
     pub fn run_with_timeout(&mut self, milliseconds: i32) -> Result<()> {
-        let event_count = self
-            .epoll_context
-            .epoll
-            .wait(
-                self.ready_events.len(),
-                milliseconds,
-                &mut self.ready_events[..],
-            )
-            .map_err(Error::Epoll)?;
+        let event_count = match self.epoll_context.epoll.wait(
```

Comment:
Might look better as:

```rust
let event_count = self.epoll_context.epoll.wait(...)
.map_err(|e| {
    if e.raw_os_error() == Some(libc::EINTR) => return Ok(());
    Error::Epoll(e)
})?;
```

Also, we should mention in the comments this may return early due to `EINTR` (I guess for `run()` as well).

Reply:
I updated the documentation. We cannot simplify the code as you suggested though :(